# Actividad M2.1 Uso de Google Colab
A01620315 David Vargas Leos

In [1]:
import pandas as pd

### Lectura de datos

In [7]:
df = pd.read_csv('titanic.csv', index_col=0,)

In [8]:
df.head()

Survived  Pclass  \
PassengerId                     
1                   0       3   
2                   1       1   
3                   1       3   
4                   1       1   
5                   0       3   

                                                          Name     Sex   Age  \
PassengerId                                                                    
1                                      Braund, Mr. Owen Harris    male  22.0   
2            Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0   
3                                       Heikkinen, Miss. Laina  female  26.0   
4                 Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0   
5                                     Allen, Mr. William Henry    male  35.0   

             SibSp  Parch            Ticket     Fare Cabin Embarked  
PassengerId                                                          
1                1      0         A/5 21171   7.2500   NaN        S  
2                1      0          PC 17599  71.2833   C85        C  
3                0      0  STON/O2. 3101282   7.9250   NaN        S  
4                1      0            113803  53.1000  C123        S  
5                0      0            373450   8.0500   NaN        S

In [9]:
df.tail()

Survived  Pclass                                      Name  \
PassengerId                                                               
887                 0       2                     Montvila, Rev. Juozas   
888                 1       1              Graham, Miss. Margaret Edith   
889                 0       3  Johnston, Miss. Catherine Helen "Carrie"   
890                 1       1                     Behr, Mr. Karl Howell   
891                 0       3                       Dooley, Mr. Patrick   

                Sex   Age  SibSp  Parch      Ticket   Fare Cabin Embarked  
PassengerId                                                                
887            male  27.0      0      0      211536  13.00   NaN        S  
888          female  19.0      0      0      112053  30.00   B42        S  
889          female   NaN      1      2  W./C. 6607  23.45   NaN        S  
890            male  26.0      0      0      111369  30.00  C148        C  
891            male  32.0      0      0      370376   7.75   NaN        Q

### Descripción de Variables 

In [10]:
categoricas = ['Name', 'Sex', 'Embarked']
numericas = ['PassengerId', 'Survived', 'Pclass', 'Age', 'SibSp', 'Parch', 'Ticket', 'Cabin']

La función "columns" te regresa el nombre de las columnas de tu dataframe

In [14]:
df.columns


Index(['Survived', 'Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Ticket',
       'Fare', 'Cabin', 'Embarked'],
      dtype='object')

La función "info" te regresa la información acerca el dataframe inlcluyendo el tipo de datos del indice y las columas, los valores NaN y el uso de memoria. 

In [13]:
df.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 891 entries, 1 to 891
Data columns (total 11 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Survived  891 non-null    int64  
 1   Pclass    891 non-null    int64  
 2   Name      891 non-null    object 
 3   Sex       891 non-null    object 
 4   Age       714 non-null    float64
 5   SibSp     891 non-null    int64  
 6   Parch     891 non-null    int64  
 7   Ticket    891 non-null    object 
 8   Fare      891 non-null    float64
 9   Cabin     204 non-null    object 
 10  Embarked  889 non-null    object 
dtypes: float64(2), int64(4), object(5)
memory usage: 83.5+ KB


La función "describe" te regresa los principales elementos de la estadística descriptiva del dataframe.

In [15]:
df.describe()

Survived      Pclass         Age       SibSp       Parch        Fare
count  891.000000  891.000000  714.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   14.526497    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   20.125000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   28.000000    0.000000    0.000000   14.454200
75%      1.000000    3.000000   38.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200

In [24]:
total_elementos = df.size
nan_elementos = df.isna().sum().sum()
no_nan_elementos = total_elementos - nan_elementos

print('Numero total de elementos: ', total_elementos)
print('Numero total de nan elementos: ', nan_elementos)
print('Numero total de no nan elementos: ', no_nan_elementos)

Numero total de elementos:  9801
Numero total de nan elementos:  866
Numero total de no nan elementos:  8935


**¿Qué columnas aparecen en el resultado de describe?** 
Survived, Pclass, Age, SibSp, Parch, Fare

**¿Cuál es la tarifa más cara?**
512.3292

**¿Cuál es el promedio de edad?**
29.6991

In [31]:
#valores_unicos_nombre = df.Name.unique()
valores_unicos_sexo = df.Sex.unique()
valores_unicos_embarked = df.Embarked.unique()

#print(valores_unicos_nombre)
print(valores_unicos_sexo)
print(valores_unicos_embarked)


['male' 'female']
['S' 'C' 'Q' nan]


### Análisis de Datos

Suma de las tarifas de todos los pasajeros

In [33]:
df.Fare.sum()

28693.9493

Datos de 5 sobrevivientes 

In [36]:
survived = df[df.Survived == 1]
survived.head()

Survived  Pclass  \
PassengerId                     
2                   1       1   
3                   1       3   
4                   1       1   
9                   1       3   
10                  1       2   

                                                          Name     Sex   Age  \
PassengerId                                                                    
2            Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0   
3                                       Heikkinen, Miss. Laina  female  26.0   
4                 Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0   
9            Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0   
10                         Nasser, Mrs. Nicholas (Adele Achem)  female  14.0   

             SibSp  Parch            Ticket     Fare Cabin Embarked  
PassengerId                                                          
2                1      0          PC 17599  71.2833   C85        C  
3                0      0  STON/O2. 3101282   7.9250   NaN        S  
4                1      0            113803  53.1000  C123        S  
9                0      2            347742  11.1333   NaN        S  
10               1      0            237736  30.0708   NaN        C

Datos de 5 no sobrevivientes

In [37]:
not_survived = df[df.Survived == 0]
not_survived.head()

Survived  Pclass                            Name   Sex   Age  \
PassengerId                                                                 
1                   0       3         Braund, Mr. Owen Harris  male  22.0   
5                   0       3        Allen, Mr. William Henry  male  35.0   
6                   0       3                Moran, Mr. James  male   NaN   
7                   0       1         McCarthy, Mr. Timothy J  male  54.0   
8                   0       3  Palsson, Master. Gosta Leonard  male   2.0   

             SibSp  Parch     Ticket     Fare Cabin Embarked  
PassengerId                                                   
1                1      0  A/5 21171   7.2500   NaN        S  
5                0      0     373450   8.0500   NaN        S  
6                0      0     330877   8.4583   NaN        Q  
7                0      0      17463  51.8625   E46        S  
8                3      1     349909  21.0750   NaN        S